### PR TITLE
Fix RTQ EQC test to model trimming correctly

### PR DESCRIPTION
**Pull request by Vagabond**
Fixed some silliness I was doing so that the test actually works right.

cc @metadave

### CODE REVIEW

File: test/riak_repl2_rtq_eqc.erl [hunk 1]
```diff
@@ -40,10 +40,9 @@ rtq_test_() ->
 
 
 max_bytes() ->
-    ?LET(MaxBytes, nat(), (MaxBytes+1) * 512).
+    ?LET(MaxBytes, nat(), (MaxBytes+1) * 39).
```
Comment (bookshelfdave) [inline test/riak_repl2_rtq_eqc.erl]: Can you comment on why 39 was chosen?

Comment (Vagabond) [inline test/riak_repl2_rtq_eqc.erl]: It is the size of the objects we are inserting into the queue (sans ETS overhead).

**Comment by Vagabond** [comment]
Ok, this is actually ready to be reviewed now.

File: test/riak_repl2_rtq_eqc.erl [hunk 2]
```diff
@@ -30,33 +31,84 @@
...
 -ifdef(TEST).
 rtq_test_() ->
-    {timeout, 30,
+    {timeout, 60,
      fun() ->
                 ?assert(eqc:quickcheck(eqc:testing_time(25,
-                                                        ?MODULE:prop_main())))
+                                                        ?MODULE:prop_main()))),
+                ?assert(eqc:quickcheck(eqc:testing_time(25,
+                                                        ?MODULE:prop_parallel())))
```
Comment (lordnull) [inline test/riak_repl2_rtq_eqc.erl]: Unloading the riak_repl_stats meck at this point prevents a large output of data due to cover being unable to find the riak_repl_stats_meck source file.

File: test/riak_repl2_rtq_eqc.erl [hunk 3]
```diff
@@ -30,33 +31,84 @@
...
         end
     }.
 -endif.
 
 
 max_bytes() ->
-    ?LET(MaxBytes, nat(), (MaxBytes+1) * 512).
+    ?LET(MaxBytes, nat(), {size, (MaxBytes+1) * 39}).
+
```
Comment (lordnull) [inline test/riak_repl2_rtq_eqc.erl:49]: Why 39?

File: test/riak_repl2_rtq_eqc.erl [hunk 4]
```diff
@@ -227,43 +312,44 @@ pull(#tc{name=Name}, Q) ->
...
-        0 ->
-            0;
-        N ->
-            choose(1, N)
-    end.
+    ?LET(E, elements(C#tc.trec), element(1, E)).
+
+trim(Q, #state{max_bytes=Max}) ->
+    {_Size, NewQ} = lists:foldl(fun({Seq, NumItems, Bin}, {Size, Acc}) ->
+                case (39 + Size) > Max of
```
Comment (lordnull) [inline test/riak_repl2_rtq_eqc.erl]: second time I've seen 39, so perhaps putting it in a descriptive macro?

**Comment by lordnull** [comment]
Has my +1, but because it has some changes from me, I'll merge it once a 3rd party has given it a quick eyeball.

**Comment by bookshelfdave** [comment]
looks good, we'll need to get this in 1.3
+1

**Pull request merged**